source/sgp_mode/HealthHost.h

```diff
@@ -7,6 +7,8 @@ class HealthHost : public SGPHost {
 
     public:
 
+    int cycles_given = 0;
```

Comment:
There is a lot of code repeated between the two PRs it looks like, which will be a problem. Each PR should be totally independent code, so if they aren't actually able to be independent, they need to be back together as one